Backend/mentii/class_ctrl.py

```diff
@@ -28,6 +28,25 @@ def getActiveClassList(dynamoDBInstance, email=None):
     response.addToPayload('classes', classes)
   return response
 
+def getClass(classCode, dynamoDBInstance, email=None, userRole=None):
+  response = ControllerResponse()
+  classData = getClassByCode(classCode, dynamoDBInstance)
+  if not classData:
+      response.addError('Get Class Failed', 'Unable to load class data')
+  else:
+    if g:
+      email = g.authenticatedUser['email']
+      userRole = g.authenticatedUser['userRole']
+    #Checks that user is the teacher of the class w/ classCode
+    if ((userRole == 'teacher' or userRole == 'admin')
```

Comment:
This logic was really rough to get through. No need to change but if others reject and you feel like breaking it up or adding to that comment I'd be happy.